DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/CentreConfiguration/CentreConfigurationControllerTests.cs

```diff
@@ -0,0 +1,130 @@
+                    .WithMockUser(true);
+
+            A.CallTo(
+                () => centresDataService.UpdateCentreWebsiteDetails(
+                    A<int>._,
+                    A<string>._,
+                    A<bool>._,
+                    A<double>._,
+                    A<double>._,
+                    A<string>._,
+                    A<string>._,
+                    A<string>._,
+                    A<string>._,
+                    A<string>._,
+                    A<string>._,
+                    A<string>._
+                )
+            ).DoesNothing();
+        }
+
+        [Test]
+        public void EditCentreWebsiteDetails_should_show_validation_error_to_user_when_given_invalid_postcode()
+        {
+            // Given
+            var model = new EditCentreWebsiteDetailsViewModel { CentrePostcode = "aaaaaaaaa" };
+            A.CallTo(() => mapsApiHelper.GeocodePostcode(A<string>._))
+                .Returns(new MapsResponse { Status = "ZERO_RESULTS" });
+
+            // When
+            var result = controller.EditCentreWebsiteDetails(model);
+
+            // Then
+            result.Should().BeViewResult().WithDefaultViewName()
+                .ModelAs<EditCentreWebsiteDetailsViewModel>();
+            Assert.IsFalse(controller.ModelState.IsValid);
+        }
+
+        [TestCase("OVER_DAILY_LIMIT")]
+        [TestCase("OVER_QUERY_LIMIT")]
+        [TestCase("REQUEST_DENIED")]
+        [TestCase("INVALID_REQUEST")]
+        [TestCase("UNKNOWN_ERROR")]
+        public void EditCentreWebsiteDetails_should_redirect_to_error_page_when_API_issue_occurs(string statusCode)
+        {
+            // Given
+            var model = new EditCentreWebsiteDetailsViewModel { CentrePostcode = "AA123" };
+            A.CallTo(() => mapsApiHelper.GeocodePostcode(A<string>._))
+                .Returns(new MapsResponse { Status = statusCode });
+
+            // When
+            var result = controller.EditCentreWebsiteDetails(model);
+
+            // Then
+            result.Should().BeRedirectToActionResult().WithActionName("Error").WithControllerName("LearningSolutions");
+        }
+
+        [Test]
+        public void EditCentreWebsiteDetails_should_show_save_coordinates_when_postcode_is_valid()
+        {
+            // Given
+            var model = new EditCentreWebsiteDetailsViewModel { CentrePostcode = "AA123" };
+            A.CallTo(() => mapsApiHelper.GeocodePostcode(A<string>._))
+                .Returns(
+                    new MapsResponse
+                    {
+                        Status = "OK",
+                        Results = new[]
+                        {
+                            new Map
+                            {
+                                Geometry = new Geometry
+                                    { Location = new Coordinates { Latitude = "50.123", Longitude = "-3.01" } }
+                            }
+                        }
+                    }

```

Comment:
Could we pull this into a private method below to keep the test more concise. Maybe something like CreateSuccessfulMapsResponse(lat, long)